warning: rowMajor2 may be used uninitialized in AdaptAutoDiff.h

<!--Please only submit issues/bug reports that come with enough information to reproduce them, ideally a unit test that fails, and possible ideas on what might be wrong. -->

<!-- Even better yet, fix the bug and/or documentation, add a unit test, and create a pull request! -->

<!-- This is a channel to report bugs/issues, not a support channel to help install/use/debug your own code. We'd love to help, but just don't have the bandwidth. Please post questions in the GTSAM Google group (https://groups.google.com/forum/#!forum/gtsam-users) -->

## Description

<!-- A clear description of the bug -->

AdaptAutoDiff.h is giving a compile warning under gcc 10.1.0.

```
In file included from /home/acxz/vcs/git/github/acxz/gtsam/gtsam/nonlinear/tests/testAdaptAutoDiff.cpp:21:
/home/acxz/vcs/git/github/acxz/gtsam/gtsam/nonlinear/AdaptAutoDiff.h: In member function ‘gtsam::AdaptAutoDiff<FUNCTOR, M, N1, N2>::VectorT gtsam::AdaptAutoDiff<FUNCTOR, M, N1, N2>::operator()(const Vector1&, const Vector2&, gtsam::OptionalJacobian<M, N1>, gtsam::OptionalJacobian<M, N2>) [with FUNCTOR = SnavelyProjection; int M = 2; int N1 = 9; int N2 = 3]’:
/home/acxz/vcs/git/github/acxz/gtsam/gtsam/nonlinear/AdaptAutoDiff.h:60:33: warning: ‘*((void*)& rowMajor2 +40)’ may be used uninitialized in this function [-Wmaybe-uninitialized]
   60 |       double rowMajor1[M * N1], rowMajor2[M * N2];  // on the stack
      |                                 ^~~~~~~~~
/home/acxz/vcs/git/github/acxz/gtsam/gtsam/nonlinear/AdaptAutoDiff.h:60:33: warning: ‘*((void*)& rowMajor2 +16)’ may be used uninitialized in this function [-Wmaybe-uninitialized]
```

## Steps to reproduce

1. git clone
2. mkdir build && cmake .. && make check

<!-- If you have a code sample, unit test, error messages, stack traces, etc., please provide it here as well -->

## Expected behavior

<!-- A clear and concise description of what you expected to happen. -->

No warning

## Environment

<!-- Please provide information about your code environment, things such as OS, language of use (C++, Matlab or Python), version of dependent libraries if using a custom build etc., anything to provide us more information. -->

<!-- The more the information, the faster we can help resolve the issue -->

OS: Arch Linux
GTSAM: latest commit on develop at time of posting
GCC: 10.1.0

## Additional information

<!-- Add any other infor or context about the problem here. -->

N/A

--- a/gtsam/nonlinear/AdaptAutoDiff.h
+++ b/gtsam/nonlinear/AdaptAutoDiff.h
@@ -57,7 +57,7 @@ class AdaptAutoDiff {
     if (H1 || H2) {
       // Get derivatives with AutoDiff
       const double* parameters[] = {v1.data(), v2.data()};
-      double rowMajor1[M * N1], rowMajor2[M * N2];  // on the stack
+      double rowMajor1[M * N1] = {}, rowMajor2[M * N2] = {};  // on the stack
       double* jacobians[] = {rowMajor1, rowMajor2};
       success = AutoDiff<FUNCTOR, double, N1, N2>::Differentiate(
           f, parameters, M, result.data(), jacobians);

